samples/simulate-ui-login.ps1

```diff
@@ -0,0 +1,143 @@
+$actionableRequests = Invoke-SafeguardMethod Core GET "Me/ActionableRequests"
+
+# GET core/v3/Me?fields=Id
+echo "GET Me.Id" 
+$userId = Invoke-SafeguardMethod Core GET "Me?fields=Id"
+
+# GET core/v3/Me/AccessRequestAssets?filter=Disabled%20eq%20false&fields=Id,Name,NetworkAddress,PlatformDisplayName&orderby=Name
+echo "GET Requestable Assets"
+$accessRequestAssets = Invoke-SafeguardMethod Core GET "Me/AccessRequestAssets?filter=Disabled%20eq%20false&fields=Id,Name,NetworkAddress,PlatformDisplayName&orderby=Name"
+
+# -- select asset --
+echo "Selecting Asset $AssetName"
+$selectedAsset = $accessRequestAssets | Where-Object Name -eq $AssetName
+if($selectedAsset -eq $null)
+{
+    throw "Specified asset is not a Requestable Asset"
+}
+
+# GET core/v3/Me/RequestEntitlements?assetIds=<num>
+echo "GET Request Entitlements"
+$requestEntitlements = Invoke-SafeguardMethod Core GET "Me/RequestEntitlements?assetIds=[$($selectedAsset.Id)]"
+
+# -- select account --
+echo "Selecting Account $AccountName"
+$selectedEntitlement = $requestEntitlements | Where-Object {$_.Account.Name -eq $AccountName}
+$selectedAccount = $selectedEntitlement.Account
+if($selectedAccount -eq $null)
+{
+    throw "Specified account is not a Requestable Account"
+}
+
+
+# POST core/v3/AccessRequests/BatchCreate
+
+$body = @( @{
+    IsEmergency = $false
+    RequestedDurationDays = 0
+    RequestedDurationHours = 10
+    RequestedDurationMinutes = 0
+    SystemId = $($selectedAsset.Id)
+    AccountId = $($selectedAccount.Id)
+    AccessRequestType = "Password"
+    })
+
+echo "POST BatchCreate"
+$batchCreateResponse = Invoke-SafeguardMethod Core POST "AccessRequests/BatchCreate" -Body $body
+
+# GET core/v3/Me/ActionableRequests
+echo "GET Actionable Requests"
+$actionableRequests = Invoke-SafeguardMethod Core GET "Me/ActionableRequests"
+
+# GET core/v3/Me/RequestableAssets/<num>/Accounts/<num>/Policies
+echo "GET Policies"
+$Policies = Invoke-SafeguardMethod Core GET "Me/RequestableAssets/$($selectedAsset.Id)/Accounts/$($selectedAccount.Id)/Policies"
+
+# POST core/v3/AccessRequests/<request_id>/CheckoutPassword
+echo "POST CheckoutPassword"
+$accountPassword = Invoke-SafeguardMethod Core POST "AccessRequests/$($batchCreateResponse.Response.Id)/CheckoutPassword"
+
+# GET core/v3/Me/ActionableRequests
+echo "GET ActionableRequests"
+$actionableRequests = Invoke-SafeguardMethod Core GET "Me/ActionableRequests"
+
+# GET core/v3/Me/RequestableAssets/<num>/Accounts/<num>/Policies
+echo "GET Policies"
+$Policies = Invoke-SafeguardMethod Core GET "Me/RequestableAssets/$($selectedAsset.Id)/Accounts/$($selectedAccount.Id)/Policies"
+
+# POST core/v3/AccessRequests/<request_id>/CheckIn
+echo "POST CheckIn"
+$checkInResponse = Invoke-SafeguardMethod Core POST "AccessRequests/$($batchCreateResponse.Response.Id)/CheckIn"
+
+# GET core/v3/Me/ActionableRequests
+echo "GET Actionable Request" 
+$actionableRequests = Invoke-SafeguardMethod Core GET "Me/ActionableRequests"
```

Comment:
Since you're using a session with this code using `Connect-Safeguard`, you should actually call `Disconnect-Safeguard` at the end.

Reply:
done